### ISSUE TITLE
Eslint_d error when .eslint.json doesn't exist.

### ISSUE THREAD

**Issue by Kazimazi** [mfussenegger/nvim-lint#462]
I have the the same issue as mentioned in https://github.com/mfussenegger/nvim-lint/issues/454#issue-1986742975 except for with **eslint_d** and solution mentioned there wouldn't work, so setting `ignore_errors = true` option for `try_lint` wouldn't work. https://github.com/mfussenegger/nvim-lint/blob/9e096df0858a1834ec2a60abe6cceb4d84dbd34c/lua/lint.lua#L208 seems to return as successful, but in the buffer I see the error:
```
Could not parse linter output due to: Expected value but found invalid token at character 1
output: Error: No Eslint configuration found in <path-to-project-root>.
```
I'm not sure how to go on debugging this issue.

It works as expected with eslint.

**Comment by tizot** [comment]
I've managed to make it work by overriding the config for that linter and adding the `--no-warn-ignored` CLI argument.
Here is an excerpt of my config with `lazy.nvim`:
```lua
{
  'mfussenegger/nvim-lint',
  event = { 'BufReadPost', 'BufNewFile' },
  opts = {
    -- other config
    linters = {
      eslint_d = {
        args = {
          '--no-warn-ignored', -- <-- this is the key argument
          '--format',
          'json',
          '--stdin',
          '--stdin-filename',
          function()
            return vim.api.nvim_buf_get_name(0)
          end,
        },
      },
    },
  },
}

**Comment by Kazimazi** [comment]
Did eslint_d lint in project with eslint config for you? `--no-warn-ignored` broke that for me.

**Comment by Camille-Upway** [comment]
> Did eslint_d lint in project with eslint config for you? `--no-warn-ignored` broke that for me.

Yes:
- in my project, `eslint` is installed as a dev dependency
- I can lint from the terminal: `npm run eslint .`
- from Neovim, the files are linted with `nvim-lint` using the `eslint_d` linter (configured as above), which in fact use the `eslint` bin installed in the repo (see https://github.com/fsouza/prettierd#local-instance)

**Comment by Kazimazi** [comment]
@Camille-Upway I'm not sure how that happens, when I try to lint from cli it would not work `eslint_d --no-warn-ignored --format json --stdin --stdin-filename jira.js < jira.js`. Keep in mind, without the `--no-warn-ignored` flag I get expected output.

**Comment by Kazimazi** [comment]
I checked how this this issue is handled in [none-ls.nvim](https://github.com/nvimtools/none-ls.nvim) and it seems that [exit code is checked](https://github.com/nvimtools/none-ls.nvim/blob/main/lua/null-ls/builtins/diagnostics/eslint.lua#L49).
My particular error has the output of:
```
Error: No ESLint configuration found in /home/kazimazi/nixos-config/home/kazimazi/features/desktop/comm
on/wayland-wm/ags/simple-bar
```
and  exit code is 2. Regular eslint rule violations would get exit code of 1 and if no errors then 0.

**Comment by jozhw** [comment]
Did anyone find a permanent solution? 

For one of my projects I made a eslintrc.json and the error disappeared. However, I did not want to keep doing that so, I opted for using the eslint-lsp with mason and lsp-zero instead.

**Comment by Kazimazi** [comment]
> Did anyone find a permanent solution?

I learned to live with it.

**Comment by KurobaneShin** [comment]
thanks to @tizot i made it


`local lint = require "lint"

local lint_augroup = vim.api.nvim_create_augroup("lint", { clear = true })

lint.linters_by_ft = {
  typescript = { "biomejs", "eslint_d", "eslint" },
  javascript = { "biomejs", "eslint_d", "eslint" },
  typescriptreact = { "biomejs", "eslint_d", "eslint" },
  javascriptreact = { "biomejs", "eslint_d", "eslint" },
}

local eslint = lint.linters.eslint_d

eslint.args = {
  "--no-warn-ignored", -- <-- this is the key argument
  "--format",
  "json",
  "--stdin",
  "--stdin-filename",
  function()
    return vim.api.nvim_buf_get_name(0)
  end,
}

vim.api.nvim_create_autocmd({ "BufWritePost" }, {
  group = lint_augroup,
  callback = function()
    lint.try_lint()
  end,
})
`

**Comment by marcus-dc** [comment]
I also got this error at my work computer and not on my PC with the same config. I did `:MasonUninstallAll`, reopened neovim and mason reinstalls all stuff within the `ensure_installed`. After that everything worked fine.

**Comment by DarhkVoyd** [comment]
A fix would be greatly appreciated!

**Comment by ewal** [comment]
Using @KurobaneShin's config but _removing_ `--no-warn-ignored` fixed it for me. Dunno what changed but it used to work having it included.

**Comment by CrzMarvin** [comment]
UPDATE: 20240814 
```lua
    eslint.args = {
      "--no-warn-ignored", -- <-- this is the key argument
      "--format",
      "json",
      "--stdin",
      "--stdin-filename",
      function()
        return vim.api.nvim_buf_get_name(0)
      end,
    }
```
above is not working, because the eslint_d's output is changed from 
'Error: No ESLint configuration found' 
to
 'Error: Could not find config file',
my current temp solution:

```lua
    lint.linters.eslint_d = require("lint.util").wrap(lint.linters.eslint_d, function(diagnostic)
      -- NOTE: 如果项目中没有配置eslint config，忽视"No ESLint configuration found"错误
      -- try to ignore "No ESLint configuration found" error
      -- update: 20240814, following is working
      -- if diagnostic.message:find("Error: No ESLint configuration found") then -- old version
      if diagnostic.message:find("Error: Could not find config file") then
        return nil
      end
      return diagnostic
    end)
```
hope to add a fix pr

**Comment by pwang2** [comment]
My case is a bit different.   I have a mono repo. if I open the UI code directory contains `.eslintrc.cjs` file or use `lcd` to change to the that directory linter working.   If I open from the mono repo root,  it will see the error.  
![image](https://github.com/user-attachments/assets/25890cd3-5def-4401-ac56-07323ed69ec6)

I assume eslint should look up the directory tree to find the closest .eslintrc in this case.

**Comment by nesterone** [comment]
Thanks, @pwang2,  I've installed 

```
MasonInstall eslint_d@13.1.2
```
 And it fixed an error with config (because eslint 8 ). However, I got another error with the parsing issue for `tsconfig`

**Comment by pwang2** [comment]
@nesterone ,  yeah!  that also works.

**Comment by arfeloreed** [comment]
> I've managed to make it work by overriding the config for that linter and adding the `--no-warn-ignored` CLI argument. Here is an excerpt of my config with `lazy.nvim`:
> 
> ```lua
> {
>   'mfussenegger/nvim-lint',
>   opts = {
>     -- other config
>     linters = {
>       eslint_d = {
>         args = {
>           '--no-warn-ignored', -- <-- this is the key argument
>           '--format',
>           'json',
>           '--stdin',
>           '--stdin-filename',
>           function()
>             return vim.api.nvim_buf_get_name(0)
>           end,
>         },
>       },
>     },
>   },
> }
> ```

This worked for me. Thanks a lot.

**Comment by Sin-cy** [comment]
> Thanks, @pwang2, I've installed
> 
> ```
> MasonInstall eslint_d@13.1.2
> ```
> 
> And it fixed an error with config (because eslint 8 )

gosh that finally fixed it

**Comment by ankitdev10** [comment]
@CrzMarvin nice catch, did the same for temporary fix.

**Comment by pwang2** [comment]
I got a chance to read some docs as I have to switch between v9 and v8.  I think the problem/confusion is clear to me now.

`eslint_d` actually supports both v8 and v9.  when a eslint package is installed to project, it will use the local node_modules installed version if exisits instead of bundled version.   It can be verified by run `!eslint_d status` in nvim. 

The confusion is in mono repo where we have node_modules folder in a child folder say UI folder. But eslint_d is using process.cwd (where we open the folder with nvim from) to detect the local install eslint.  
https://github.com/mantoni/eslint_d.js/blob/df020090106a9a887e32ed05645b020f15a04f00/lib/resolver.js#L20C1-L21C1

![image](https://github.com/user-attachments/assets/36f2fadc-fa40-4559-98ac-f6b02a0b78b9)

![image](https://github.com/user-attachments/assets/c1e4af99-d4da-4dbc-ad7f-1459b684d373)

TLDR: with mono repo, just cd to the folder where there is a desired version eslint in that directory's node_modules folder.   MasonInstall to Mason's binary path should be avoided.

**Comment by Marat-Gumerov** [comment]
> I got a chance to read some docs as I have to switch between v9 and v8. I think the problem/confusion is clear to me now.
> 
> `eslint_d` actually supports both v8 and v9. when a eslint package is installed to project, it will use the local node_modules installed version if exisits instead of bundled version. It can be verified by run `!eslint_d status` in nvim.
> 
> The confusion is in mono repo where we have node_modules folder in a child folder say UI folder. But eslint_d is using process.cwd (where we open the folder with nvim from) to detect the local install eslint. https://github.com/mantoni/eslint_d.js/blob/df020090106a9a887e32ed05645b020f15a04f00/lib/resolver.js#L20C1-L21C1
> 
> ![image](https://private-user-images.githubusercontent.com/1592014/380291082-36f2fadc-fa40-4559-98ac-f6b02a0b78b9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAyMTQ4NzYsIm5iZiI6MTczMDIxNDU3NiwicGF0aCI6Ii8xNTkyMDE0LzM4MDI5MTA4Mi0zNmYyZmFkYy1mYTQwLTQ1NTktOThhYy1mNmIwMmEwYjc4YjkucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTAyOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEwMjlUMTUwOTM2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YTU0YmIxYThlZTY1MTllZWFmNTMxYjI3MTA2NDdlYWUyNjg3OGM4OWU5NWVhY2JhMzA3YzkzNjEyMjNkOGEzYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.aHmTGeP5dbjZXsz4xcEP4W5bhNk1rtCtzFbhNhVdm6I)
> 
> ![image](https://private-user-images.githubusercontent.com/1592014/380291368-c1e4af99-d4da-4dbc-ad7f-1459b684d373.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAyMTQ4NzYsIm5iZiI6MTczMDIxNDU3NiwicGF0aCI6Ii8xNTkyMDE0LzM4MDI5MTM2OC1jMWU0YWY5OS1kNGRhLTRkYmMtYWQ3Zi0xNDU5YjY4NGQzNzMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTAyOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEwMjlUMTUwOTM2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MTBlYmY5NWFiNTg4ZmIzNDRkMDEzMTZkOTY0MWNlNTNkZWViNWExMWE0ZGY0NjhjMTU5YmQ2MTFmMzU1MGQ2YiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.J05iNO4Bb8vfrGarCrZkAlNTAgLCjQDDEDoPyg0siGo)
> 
> TLDR: with mono repo, just cd to the folder where there is a desired version eslint in that directory's node_modules folder. MasonInstall to Mason's binary path should be avoided.

```lua
        local function find_nearest_node_modules_dir()
            -- current buffer dir
            local current_dir = vim.fn.expand('%:p:h')
            while current_dir ~= "/" do
                if vim.fn.isdirectory(current_dir .. "/node_modules") == 1 then
                    return current_dir
                end
                current_dir = vim.fn.fnamemodify(current_dir, ":h")
            end
            return nil
        end

        vim.api.nvim_create_autocmd({ "BufEnter", "BufWritePost", "InsertLeave" }, {
            group = lint_augroup,
            callback = function()
                local ft = vim.bo.filetype
                local js_types = { "javascript", "typescript", "javascriptreact", "typescriptreact" }
                if not vim.tbl_contains(js_types, ft) then
                    lint.try_lint()
                    return
                end
                local original_cwd = vim.fn.getcwd()
                local node_modules_dir = find_nearest_node_modules_dir()
                if node_modules_dir then
                    vim.cmd("cd " .. node_modules_dir)
                end
                lint.try_lint()
                vim.cmd("cd " .. original_cwd)
            end,
        })
```

**Comment by pierskarsenbarg** [comment]
Just wanted to add for next time I'm looking for this, if you're using `WhoIsSethDaniel/mason-tool-installer.nvim`, you can specify the version so you don't have to run a command:

```lua
require('mason-tool-installer').setup {
    ensure_installed =  {
        ...
        {'eslint_d', version = '13.1.2'}, // this is the important line
        ...
    })
}
```

**Comment by ajay-david-goggins** [comment]
1 Diagnostics: 2 1. failed to decode json: Expected value but found invalid token at character 1 // i got this problem 

my lspconfig.lua file is 
return { -- Mason Installer
    {
        "williamboman/mason.nvim",
        config = function()
            require("mason").setup()
        end,
    },

    -- Mason LSP Installer
    {
        "williamboman/mason-lspconfig.nvim",
        config = function()
            require("mason-lspconfig").setup({
                ensure_installed = {
                    "lua_ls",
                    "ts_ls", -- Γ£à ts_ls instead of tsserver
                    "jdtls",
                    "html",
                    "cssls",
                    "clangd",
                    "emmet_ls",
                    "angularls",
                    "eslint",
                },
            })
        end,
    },

    -- DAP Support
    {
        "jay-babu/mason-nvim-dap.nvim",
        config = function()
            require("mason-nvim-dap").setup({
                ensure_installed = { "java-debug-adapter", "java-test" },
            })
        end,
    },

    -- Java LSP (JDTLS)
    {
        "mfussenegger/nvim-jdtls",
        dependencies = { "mfussenegger/nvim-dap" },
    },

    -- LSP Setup
    {
        "neovim/nvim-lspconfig",
        config = function()
            local lspconfig = require("lspconfig")
            local capabilities = require("cmp_nvim_lsp").default_capabilities()
            local util = require("lspconfig.util")

            lspconfig.lua_ls.setup({ capabilities = capabilities })
            lspconfig.ts_ls.setup({ capabilities = capabilities }) -- Γ£à ts_ls setup here
            lspconfig.html.setup({ capabilities = capabilities })
            lspconfig.cssls.setup({ capabilities = capabilities })
            lspconfig.clangd.setup({ capabilities = capabilities })

            -- Emmet
            lspconfig.emmet_ls.setup({
                capabilities = capabilities,
                filetypes = {
                    "html", "css", "php", "blade",
                    "javascriptreact", "typescriptreact",
                    "astro", "vue", "svelte",
                },
                init_options = {
                    html = {
                        options = {
                            ["bem.enabled"] = true,
                        },
                    },
                },
            })

            -- Angular
            lspconfig.angularls.setup({
                capabilities = capabilities,
                root_dir = util.root_pattern("angular.json", "package.json", "nx.json", ".git"),
                filetypes = { "typescript", "html", "scss", "css" },
                settings = {
                    angular = {
                        strictTemplates = true,
                    },
                },
            })

            -- ESLint
            lspconfig.eslint.setup({
                capabilities = capabilities,
                on_attach = function(_, bufnr)
                    vim.api.nvim_create_autocmd("BufWritePre", {
                        buffer = bufnr,
                        command = "EslintFixAll",
                    })
                end,
                settings = {
                    workingDirectory = { mode = "auto" },
                    experimental = {
                        useFlatConfig = true,
                    },
                },
                root_dir = util.root_pattern(
                    "eslint.config.mjs",
                    ".eslintrc",
                    ".eslintrc.js",
                    ".eslintrc.json",
                    ".eslintrc.cjs",
                    "package.json",
                    ".git"
                ),
            })

            -- Keymaps
            vim.keymap.set("n", "<leader>ch", vim.lsp.buf.hover, { desc = "[C]ode [H]over" })
            vim.keymap.set("n", "<leader>cd", vim.lsp.buf.definition, { desc = "[C]ode [D]efinition" })
            vim.keymap.set({ "n", "v" }, "<leader>ca", vim.lsp.buf.code_action, { desc = "[C]ode [A]ction" })
            vim.keymap.set("n", "<leader>cr", require("telescope.builtin").lsp_references, { desc = "[C]ode [R]eferences" })
            vim.keymap.set("n", "<leader>ci", require("telescope.builtin").lsp_implementations, { desc = "[C]ode [I]mplementations" })
            vim.keymap.set("n", "<leader>cR", vim.lsp.buf.rename, { desc = "[C]ode [R]ename" })
            vim.keymap.set("n", "<leader>cD", vim.lsp.buf.declaration, { desc = "[C]ode [D]eclaration" })
        end,
    },

    -- Null-ls + Mason Integration
    {
        "nvimtools/none-ls.nvim",
        dependencies = {
            "williamboman/mason.nvim",
            "jay-babu/mason-null-ls.nvim",
        },
        config = function()
            require("mason-null-ls").setup({
                ensure_installed = {
                    "eslint_d",
                    "prettier",
                },
                automatic_installation = true,
            })

            local null_ls = require("null-ls")

            null_ls.setup({
                sources = {
                    null_ls.builtins.diagnostics.eslint_d.with({
                        condition = function(utils)
                            return utils.root_has_file({
                                "eslint.config.mjs",
                                ".eslintrc",
                                ".eslintrc.json",
                                "package.json",
                            })
                        end,
                    }),
                    null_ls.builtins.formatting.prettier.with({
                        condition = function(utils)
                            return utils.root_has_file({
                                ".prettierrc",
                                ".prettierrc.js",
                                "package.json",
                            })
                        end,
                    }),
                },
            })
        end,
    },
}

in my project i use the 
   28   suck
   27     .vscode
   26     node_modules
   25     public
   24     src
   23       app
   22          app.config.server.ts
   21          app.config.ts
   20          app.html
   19          app.routes.server.ts
   18          app.routes.ts
   17          app.scss
   16          app.spec.ts
   15          app.ts
   14        index.html
   13        main.server.ts
   12        main.ts
   11        server.ts
   10        styles.scss
    9      .editorconfig
    8      .gitignore
    7      angular.json
    6      eslint.config.mjs             <========== this type of file i used ==========================<<<
    5      package-lock.json
    4      package.json
    3     󰂺 README.md
    2      tsconfig.app.json
    1      tsconfig.json
  87       tsconfig.spec.json

I try lot of time with differetn. i am begginer for this. 

any one help me with that ?

**Comment by rauf322** [comment]
I tried all the methods above what i comes up to use noice from folke and put filter to not disturb me with my eslint error 

```lua
//"folke/noice.nvim",
{

filter = {
	event = "notify",
        find = "ESLint output parsing failed",
        },
       opts = { skip = true },
},
```
above method works with noice.nvim feel free to have a look at the full code in my dotfiles

And then i remove the problem with LSP by using just parsing error and remove it our 

```lua
			lint.linters.eslint_d = {
				cmd = "eslint_d",
				stdin = true,
				args = {
					"--format",
					"json",
					"--stdin",
					"--stdin-filename",
					function()
						return vim.api.nvim_buf_get_name(0)
					end,
				},
				stream = "stdout",
				ignore_exitcode = true,
				parser = function(output, bufnr)
					if output == "" then
						return {}
					end

					local ok, result = pcall(vim.json.decode, output)
					if not ok then
						vim.notify("ESLint output parsing failed: " .. tostring(result), vim.log.levels.WARN)
						return {}
					end

					if not result or not result[1] or not result[1].messages then
						return {}
					end

					local diagnostics = {}
					for _, message in ipairs(result[1].messages) do
						table.insert(diagnostics, {
							lnum = message.line - 1,
							col = message.column - 1,
							end_lnum = message.endLine and (message.endLine - 1) or (message.line - 1),
							end_col = message.endColumn and (message.endColumn - 1) or (message.column - 1),
							severity = message.severity == 1 and vim.diagnostic.severity.WARN
								or vim.diagnostic.severity.ERROR,
							message = message.message,
							source = "eslint_d",
							code = message.ruleId,
						})
					end

					return diagnostics
				end,
			}

```


The problem is super annoying, i hope that will be helpful
